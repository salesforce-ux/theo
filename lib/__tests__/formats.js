// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

const path = require('path')

const theo = require('../')

const getFixture = file =>
  path.resolve(__dirname, '../__fixtures__/', file)

const format = (transform, format, fixture = 'a.json') =>
  theo
    .transform({
      type: transform,
      file: getFixture(fixture)
    })
    .chain(data =>
      theo.format({
        type: format,
        data
      })
    )
    .merge()

describe('formats', () => {
  it('raw.json', () => {
    expect(format('raw', 'raw.json')).toMatchSnapshot()
  })
  it('android.xml', () => {
    expect(format('android', 'android.xml')).toMatchSnapshot()
  })
  it('aura.tokens', () => {
    expect(format('aura', 'aura.tokens')).toMatchSnapshot()
  })
  it('common.js', () => {
    expect(format('web', 'common.js')).toMatchSnapshot()
  })
  it('html', () => {
    expect(format('web', 'html')).toMatchSnapshot()
  })
  it('less', () => {
    expect(format('web', 'less')).toMatchSnapshot()
  })
  it('list.scss', () => {
    expect(format('web', 'list.scss')).toMatchSnapshot()
  })
  it('map.scss', () => {
    expect(format('web', 'map.scss')).toMatchSnapshot()
  })
  it('map.variables.scss', () => {
    expect(format('web', 'map.variables.scss')).toMatchSnapshot()
  })
  it('sass.default', () => {
    expect(format('web', 'sass.default')).toMatchSnapshot()
  })
  it('sass', () => {
    expect(format('web', 'sass')).toMatchSnapshot()
  })
  it('scss.default', () => {
    expect(format('web', 'scss.default')).toMatchSnapshot()
  })
  it('scss', () => {
    expect(format('web', 'scss')).toMatchSnapshot()
  })
  it('styl', () => {
    expect(format('web', 'styl')).toMatchSnapshot()
  })
})
