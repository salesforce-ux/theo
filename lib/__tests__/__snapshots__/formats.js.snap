// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`android.xml 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<resources>
  <property name=\\"TOKEN_COLOR\\" category=\\"background-color\\">#bada55</property>
  <property name=\\"TOKEN_SIZE\\" category=\\"spacing\\">20px</property>
  <property name=\\"TOKEN_STRING\\" category=\\"test\\">/assets/images</property>
  <property name=\\"TOKEN_NUMBER\\" category=\\"test\\">2</property>
  <property name=\\"TOKEN_QUOTES\\" category=\\"test\\">&apos;Salesforce Sans&apos;, &quot;Helvetica Neue&quot;, sans-serif</property>
</resources>"
`;

exports[`aura.tokens (exclude attrs) 1`] = `
"<aura:tokens>
  <aura:token name=\\"tokenColor\\" value=\\"#bada55\\"/>
  <aura:token name=\\"tokenSize\\" value=\\"20px\\"/>
  <aura:token name=\\"tokenString\\" value=\\"/assets/images\\"/>
  <aura:token name=\\"tokenNumber\\" value=\\"2\\"/>
  <aura:token name=\\"tokenQuotes\\" value=\\"&apos;Salesforce Sans&apos;, &quot;Helvetica Neue&quot;, sans-serif\\"/>
  <aura:token name=\\"tokenCss\\" value=\\"white\\"/>
</aura:tokens>"
`;

exports[`aura.tokens 1`] = `
"<aura:tokens extends=\\"theme\\">
  <aura:import name=\\"importOne\\"/>
  <aura:import name=\\"importTwo\\"/>
  <aura:token name=\\"tokenColor\\" value=\\"#bada55\\"/>
  <aura:token name=\\"tokenSize\\" value=\\"20px\\"/>
  <aura:token name=\\"tokenString\\" value=\\"/assets/images\\"/>
  <aura:token name=\\"tokenNumber\\" value=\\"2\\"/>
  <aura:token name=\\"tokenQuotes\\" value=\\"&apos;Salesforce Sans&apos;, &quot;Helvetica Neue&quot;, sans-serif\\"/>
  <aura:token name=\\"tokenCss\\" property=\\"background,border\\" value=\\"white\\"/>
</aura:tokens>"
`;

exports[`common.js 1`] = `
"module.exports = {
  tokenColor: \\"#bada55\\",
  tokenSize: \\"20px\\",
  tokenString: \\"/assets/images\\",
  tokenNumber: 2,
  // This should not get escaped in the output
  tokenQuotes: \\"'Salesforce Sans', \\\\\\"Helvetica Neue\\\\\\", sans-serif\\",
};"
`;

exports[`cssmodules.css 1`] = `
"
@value token-color: #bada55;
@value token-size: 20px;
@value token-string: \\"/assets/images\\";
@value token-number: 2;
/* This should not get escaped in the output */
@value token-quotes: 'Salesforce Sans', \\"Helvetica Neue\\", sans-serif;
"
`;

exports[`custom-properties.css 1`] = `
"
:root {
  --token-color: #bada55;
  --token-size: 20px;
  --token-string: \\"/assets/images\\";
  --token-number: 2;
  /* This should not get escaped in the output */
  --token-quotes: 'Salesforce Sans', \\"Helvetica Neue\\", sans-serif;
}
"
`;

exports[`default.sass 1`] = `
"
$token-color: #bada55 !default
$token-size: 20px !default
$token-string: \\"/assets/images\\" !default
$token-number: 2 !default
// This should not get escaped in the output
$token-quotes: 'Salesforce Sans', \\"Helvetica Neue\\", sans-serif !default
"
`;

exports[`default.scss 1`] = `
"
$token-color: #bada55 !default;
$token-size: 20px !default;
$token-string: \\"/assets/images\\" !default;
$token-number: 2 !default;
// This should not get escaped in the output
$token-quotes: 'Salesforce Sans', \\"Helvetica Neue\\", sans-serif !default;
"
`;

exports[`html 1`] = `
"
      <!DOCTYPE html>
      <html lang=\\"en\\">
        <head>
          <meta charset=\\"utf-8\\" />
          <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\" />
          <title>Design Tokens</title>
          <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\" />
          <link rel=\\"stylesheet\\" href=\\"https://cdnjs.cloudflare.com/ajax/libs/normalize/4.2.0/normalize.min.css\\" />
          <style>
            html {
              box-sizing: border-box;
              font-size: 1em;
              font-family: Helvetica, Arial, sans-serif;
              line-height: 1.5;
              background: #fff;
              color: #444;
            }
            *,
            *:before,
            *:after {
              box-sizing: inherit;
            }
            body { font-size: .75rem; }
            ::-moz-selection {
              background: #b3d4fc;
              text-shadow: none;
            }
            ::selection {
              background: #b3d4fc;
              text-shadow: none;
            }
            .banner,
            .contentinfo { background: #f5f5f5; }
            .banner { padding: 1em 0; }
            .container {
              margin: 0 auto;
              padding: 0 2rem;
              max-width: 80rem;
            }
            main {
              margin-right: -2rem;
              margin-left:  -2rem;
              padding-bottom: 2rem;
            }
            h1 {
              margin: 0;
              font-weight: normal;
              line-height: 1.25;
            }
            table {
              table-layout: fixed;
              border-collapse: separate;
              border-spacing: 1rem;
              width: 100%;
            }
            th,
            td {
              padding: 0 1rem;
              vertical-align: baseline;
              word-break: break-word;
              hyphens: auto;
            }
            th {
              font-weight: normal;
              text-align: left;
            }
            thead th {
              border-bottom: 1px solid #eee;
              padding-top: 1rem;
              padding-bottom: .5rem;
              color: #999;
            }
            thead th:first-child {
              font-size: 1.25rem;
              color: inherit;
            }
            code { font-family: monaco, Consolas, monospace, monospace; }
            hr {
              display: block;
              margin: 2rem 0;
              border: 0;
              border-top: 1px solid #eee;
              padding: 0;
              height: 1px;
            }
            .metric-box,
            .radius-box {
              display: inline-block;
              vertical-align: middle;
              background: #eee;
            }
            .radius-box {
              width: 100%;
              height: 3rem;
            }
            .radius-box.borderRadiusCircle {
              width: 3rem;
            }
            .line-height-example {
              border-bottom: 1px solid #eee;
              background-image: -webkit-linear-gradient(#eee 1px, transparent 1px);
              background-image: linear-gradient(#eee 1px, transparent 1px);
              background-size: 100% 1rem;
            }
          </style>
        </head>
        <body>
          <header role=\\"banner\\" class=\\"banner\\">
            <div class=\\"container\\">
              <h1>Design Tokens</h1>
            </div>
          </header>
          <div class=\\"container\\">
            <main role=\\"main\\">
              
              
      <section>
        <table>
          
      <thead>
        <tr id=spacing>
          <th scope=\\"col\\">Spacing</th>
          <th scope=\\"col\\">Value</th>
          <th scope=\\"col\\">Examples</th>
          <th scope=\\"col\\">Usage</th>
        </tr>
      </thead>
    
          <tbody>
            <tr>
        <th scope=\\"row\\">
          <code>tokenSize</code>
        </th>
        <td>
          <code>20px</code>
        </td>
        
        <td>
          <div class=\\"metric-box\\" style=\\"width: 20px; height: 20px;\\"></div>
        </td>
      
        <td></td>
      </tr>
          </tbody>
        </table>
        <hr />
      </section>
    
              
              
              
              
              
              
              
              
              
              
              
              
      <section>
        <table>
          
      <thead>
        <tr id=background-color>
          <th scope=\\"col\\">Background Colors</th>
          <th scope=\\"col\\">Value</th>
          <th scope=\\"col\\">Examples</th>
          <th scope=\\"col\\">Usage</th>
        </tr>
      </thead>
    
          <tbody>
            <tr>
        <th scope=\\"row\\">
          <code>tokenColor</code>
        </th>
        <td>
          <code>#bada55</code>
        </td>
        <td style=\\"background-color: #bada55; border: 1px solid #f6f6f6;\\"></td>
        <td></td>
      </tr>
          </tbody>
        </table>
        <hr />
      </section>
    
              
              
              
              
              
              
              
              
              
            </main>
          </div>
        </body>
      </html>
    "
`;

exports[`ios.json 1`] = `
"{
  \\"properties\\": [
    {
      \\"category\\": \\"background-color\\",
      \\"type\\": \\"color\\",
      \\"name\\": \\"tokenColor\\",
      \\"value\\": \\"#bada55\\"
    },
    {
      \\"category\\": \\"spacing\\",
      \\"type\\": \\"size\\",
      \\"name\\": \\"tokenSize\\",
      \\"value\\": \\"20px\\"
    },
    {
      \\"category\\": \\"test\\",
      \\"type\\": \\"string\\",
      \\"name\\": \\"tokenString\\",
      \\"value\\": \\"/assets/images\\"
    },
    {
      \\"category\\": \\"test\\",
      \\"type\\": \\"number\\",
      \\"name\\": \\"tokenNumber\\",
      \\"value\\": 2
    },
    {
      \\"category\\": \\"test\\",
      \\"type\\": \\"test\\",
      \\"name\\": \\"tokenQuotes\\",
      \\"value\\": \\"'Salesforce Sans', \\\\\\"Helvetica Neue\\\\\\", sans-serif\\",
      \\"comment\\": \\"This should not get escaped in the output\\"
    }
  ]
}"
`;

exports[`json 1`] = `
"{
  \\"TOKEN_COLOR\\": \\"#bada55\\",
  \\"TOKEN_SIZE\\": \\"20px\\",
  \\"TOKEN_STRING\\": \\"/assets/images\\",
  \\"TOKEN_NUMBER\\": 2,
  \\"TOKEN_QUOTES\\": \\"'Salesforce Sans', \\\\\\"Helvetica Neue\\\\\\", sans-serif\\"
}"
`;

exports[`less 1`] = `
"
@token-color: #bada55;
@token-size: 20px;
@token-string: \\"/assets/images\\";
@token-number: 2;
// This should not get escaped in the output
@token-quotes: 'Salesforce Sans', \\"Helvetica Neue\\", sans-serif;
"
`;

exports[`list.scss 1`] = `
"
$fixture-list: (
  \\"token-color\\",
  \\"token-size\\",
  \\"token-string\\",
  \\"token-number\\",
  // This should not get escaped in the output
  \\"token-quotes\\",
);
"
`;

exports[`map.scss 1`] = `
"
$fixture-map: (
  'token-color': (#bada55),
  'token-size': (20px),
  'token-string': (\\"/assets/images\\"),
  'token-number': (2),
  // This should not get escaped in the output
  'token-quotes': ('Salesforce Sans', \\"Helvetica Neue\\", sans-serif),
);
"
`;

exports[`map.variables.scss 1`] = `
"
$fixture-map: (
  'token-color': $token-color,
  'token-size': $token-size,
  'token-string': $token-string,
  'token-number': $token-number,
  // This should not get escaped in the output
  'token-quotes': $token-quotes,
);
"
`;

exports[`module.js 1`] = `
"export const tokenColor = \\"#bada55\\";
export const tokenSize = \\"20px\\";
export const tokenString = \\"/assets/images\\";
export const tokenNumber = 2;
// This should not get escaped in the output
export const tokenQuotes = \\"'Salesforce Sans', \\\\\\"Helvetica Neue\\\\\\", sans-serif\\";"
`;

exports[`raw.json 1`] = `
"{
  \\"aliases\\": {},
  \\"props\\": {
    \\"TOKEN_COLOR\\": {
      \\"category\\": \\"background-color\\",
      \\"type\\": \\"color\\",
      \\"name\\": \\"TOKEN_COLOR\\",
      \\"value\\": \\"#bada55\\",
      \\"originalValue\\": \\"#bada55\\"
    },
    \\"TOKEN_SIZE\\": {
      \\"category\\": \\"spacing\\",
      \\"type\\": \\"size\\",
      \\"name\\": \\"TOKEN_SIZE\\",
      \\"value\\": \\"20px\\",
      \\"originalValue\\": \\"20px\\"
    },
    \\"TOKEN_STRING\\": {
      \\"category\\": \\"test\\",
      \\"type\\": \\"string\\",
      \\"name\\": \\"TOKEN_STRING\\",
      \\"value\\": \\"/assets/images\\",
      \\"originalValue\\": \\"/assets/images\\"
    },
    \\"TOKEN_NUMBER\\": {
      \\"category\\": \\"test\\",
      \\"type\\": \\"number\\",
      \\"name\\": \\"TOKEN_NUMBER\\",
      \\"value\\": 2,
      \\"originalValue\\": 2
    },
    \\"TOKEN_QUOTES\\": {
      \\"category\\": \\"test\\",
      \\"type\\": \\"test\\",
      \\"name\\": \\"TOKEN_QUOTES\\",
      \\"value\\": \\"'Salesforce Sans', \\\\\\"Helvetica Neue\\\\\\", sans-serif\\",
      \\"comment\\": \\"This should not get escaped in the output\\",
      \\"originalValue\\": \\"'Salesforce Sans', \\\\\\"Helvetica Neue\\\\\\", sans-serif\\"
    }
  }
}"
`;

exports[`sass 1`] = `
"
$token-color: #bada55
$token-size: 20px
$token-string: \\"/assets/images\\"
$token-number: 2
// This should not get escaped in the output
$token-quotes: 'Salesforce Sans', \\"Helvetica Neue\\", sans-serif
"
`;

exports[`scss 1`] = `
"
$token-color: #bada55;
$token-size: 20px;
$token-string: \\"/assets/images\\";
$token-number: 2;
// This should not get escaped in the output
$token-quotes: 'Salesforce Sans', \\"Helvetica Neue\\", sans-serif;
"
`;

exports[`styl 1`] = `
"
$token-color = #bada55
$token-size = 20px
$token-string = \\"/assets/images\\"
$token-number = 2
// This should not get escaped in the output
$token-quotes = 'Salesforce Sans', \\"Helvetica Neue\\", sans-serif
"
`;
